source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -308,28 +308,71 @@ ShaderNodePtr ShaderNode::create(const string& name, const NodeDef& nodeDef, Sha
     return newNode;
 }
 
-void ShaderNode::setValues(const Node& node, const NodeDef& nodeDef, ShaderGenerator& shadergen)
+void ShaderNode::setElementPaths(const Node& node, const NodeDef& nodeDef, bool includeNodeDefInputs)
 {
-    // Copy input values from the given node
-    const vector<ValueElementPtr> nodeInputs = node.getChildrenOfType<ValueElement>();
+    // Set element paths for children on the node
+    const vector<ValueElementPtr> nodeValues = node.getChildrenOfType<ValueElement>();
+    for (const ValueElementPtr& nodeValue : nodeValues)
+    {
+        ShaderInput* input = getInput(nodeValue->getName());
+        if (input)
+        {
+            input->elementPath = nodeValue->getNamePath();
+        }
+    }
+
+    if (!includeNodeDefInputs)
+    {
+        return;
+    }
+
+    // Set element paths based on the node definition. Note that these
+    // paths don't actually exist at time of shader generation since there
+    // are no inputs/parameters specified on the node itself
+    //
```

Comment:
It's not a path to a nodedef but a path to a node input or parameter if it existed. So if at some point the input or parameter is added as a child of the node then this path is usable to get those elements. Otherwise searching for them will return a failure.

If a nodedef itself did change then a rebuild would probably be required.

Reply:
Sorry my bad, I misread the code. This looks good!